api/user/models.py

```diff
@@ -19,7 +19,7 @@ class User(Base, Utility):
     __tablename__ = 'users'
     id = Column(Integer, Sequence('users_id_seq'), primary_key=True)
     email = Column(String, nullable=False, unique=True)
-    location = Column(String, nullable=True)
+    location = Column(String, default="Kampala")
```

Comment:
@nakatuddesuzan kindly shade some light as to why we have a default location as Kampala

Reply:
Okay, so if you follow up the past conversation, to do away with the decorator, we had to find a way of making sure whenever the tests run, the location of the user is saved. So this is one way that could be fixed. So if the user has no location, Kampala is saved as the location.

Reply:
In the production environment, kindly let us know situations when a user would not have their location when setting up their Converge accounts for the first time, and why we need to make Kampala their default location hence the need to have this set in the database user model.

Reply:
Hello @Bonifase , Here I was solving the tests problem. Following the recommendation, you gave for removing the decorator. Since the analytics view is for a specific location, the tests were failing because everytime the tests run, the location of the user is not saved in the test database. The decorator was solving this problem because the developer who designed it set a location of the user to Nairobi as a default. So if the decorator is removed, it then means, there is no way to get the user location in the test database. The other way I could implement it would be passing the location parameter in the init method in the user model class, but this would call for a change in the authentication implementation since the tokens of the user missed out the location information when they were being encoded. So by setting the default location would solve the problem. this is only affecting the test database. I chose Kampala because the offices being created while the tests run are in Kampala office.

Reply:
The implementaion of removing the decorator works well and on insomnia, both users can view analytics because users are saved with their locations. Problem comes when tests run the user location is not saved, so the view analytics tests can not pass.